src/utils/presets/migrateTreasuryFunds.js

```diff
@@ -0,0 +1,145 @@
+import { ethers } from 'ethers'
+import { erc20ABI } from 'wagmi'
+import { readContract } from '@wagmi/core'
+
+import { getEthPNTAsset, getRawAmount, pNetworkV2Vault, vaultContract } from './utils'
+import settings from '../../settings'
+
+const migrateTreasuryFunds = ({ presetParams, setPresetParams }) => ({
+  id: 'migrateTreasuryFunds',
+  name: 'Migrate Treasury Funds',
+  description: 'Pegin Treasury funds to new chain',
+  args: [
+    {
+      id: 'select-token-address',
+      name: 'tokenAddress',
+      component: 'AssetSelection',
+      props: {
+        assets: settings.assets.filter(
+          (asset) => asset.symbol === 'pBTC' || asset.symbol === 'ethPNT' || asset.symbol === 'PNT'
+        ),
+        onSelect: (_address) =>
+          setPresetParams({
+            ...presetParams,
+            0: _address
+          })
+      }
+    },
+    {
+      id: 'input-amount',
+      name: 'amount',
+      component: 'Input',
+      props: {
+        type: 'number',
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Amount ...',
+        value: presetParams[1] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            1: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'select-chain',
+      name: 'destinationChain',
+      component: 'ChainSelection',
+      props: {
+        onSelect: (_chainId) =>
+          setPresetParams({
+            ...presetParams,
+            2: _chainId
+          })
+      }
+    },
+    {
+      id: 'input-receiver-address',
+      name: 'receiverAddress',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'New Treasury Address (receiver address) ...',
+        value: presetParams[3] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            3: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'input-user-data',
+      name: 'userData',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Optional User Data ...',
+        value: presetParams[4] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            4: _e.target.value
+          })
+      }
+    }
+  ],
+  prepare: async () => {
+    if (Object.values(presetParams).length < 2) return null
+    const assetAddress = presetParams[0] ? presetParams[0] : getEthPNTAsset().address
+    const amount = presetParams[1]
+    const destinationChainId = !presetParams[2] ? settings.chains[0].chainId : presetParams[2]
```

Comment:
I just noticed this: destinationChainId should by the usual bytes4 we use in pNetwork v2. I doubt settings.chains[i].chainId is that metadata network ID.

Reply:
it is! https://github.com/pnetwork-association/dao-v2-ui/blob/feat/migrateTreasuryFunds/src/settings/index.js#L96
I'll add a test to explicitly cover this (as it is for other similar preset)

Reply:
I see! all good then